### PR TITLE
fix: headers are immutable error

### LINKED ISSUE #14: Headers are immutable error
The error is caused by https://github.com/satyarohith/sift/blob/fd0f5862e877fafbb08b7ca59a28546571743b00/mod.ts#L176

This throws an error because `fetch()` is more spec compliant from `1.9.1` and headers are immutable.

(Thanks @timonson for reporting!)

### PR DESCRIPTION
**Pull request by ngruychev**
closes #14
this PR adds a helper function to create a new Response with additional headers instead of trying to set headers on the original Response

### CODE REVIEW

File: mod.ts [hunk 1]
```diff
@@ -57,6 +57,17 @@ export function serve(userRoutes: Routes): void {
   });
 }
 
+async function newResponse(res: Response, headers: Record<string, string>): Promise<Response> {
+    return new Response(await res.blob(), {
```
Comment (satyarohith) [inline mod.ts:61]:
```suggestion
    return new Response(await res.body, {
```

`blob()` might not work for streaming responses.

Comment (ngruychev) [reply mod.ts]: Thanks for the correction!

**Review by satyarohith — CHANGES_REQUESTED** [review]
Everything looks good apart from the lint error. :)

**Comment by ngruychev** [comment]
lint error fixed :)

**Review by satyarohith — APPROVED** [review]
Thanks, @ngruychev!

I'll fix the test - it requires using the new deploy runtime polyfill.

**Pull request merged**